# 1.0 `About Dataset`
Context
While many public datasets (on Kaggle and the like) provide Apple App Store data, there are not many counterpart datasets available for Google Play Store apps anywhere on the web. On digging deeper, I found out that iTunes App Store page deploys a nicely indexed appendix-like structure to allow for simple and easy web scraping. On the other hand, Google Play Store uses sophisticated modern-day techniques (like dynamic page load) using JQuery making scraping more challenging.

### 1.1 Content
Each app (row) has values for catergory, rating, size, and more.

### 1.2 Acknowledgements
This information is scraped from the Google Play Store. This app information would not be available without it.

### 1.3 Inspiration
The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market!

### 1.4 Dataset link:
> The Data Set was downloaded from Kaggle, from the following [link](https://www.kaggle.com/datasets/lava18/google-play-store-apps/)

# 2.0 `Import Libraries:`

In [2]:
# import libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# 3.0 `Exploring Data:`

In [3]:
# load data
df = pd.read_csv('../../datasets/googleplaystore.csv')

In [4]:
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

   Reviews  Size     Installs  Type Price Content Rating  \
0      159   19M      10,000+  Free     0       Everyone   
1      967   14M     500,000+  Free     0       Everyone   
2    87510  8.7M   5,000,000+  Free     0       Everyone   
3   215644   25M  50,000,000+  Free     0           Teen   
4      967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [5]:
df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating  Reviews                Size     Installs  Type Price  \
10836     4.5       38                 53M       5,000+  Free     0   
10837     5.0        4                3.6M         100+  Free     0   
10838     NaN        3                9.5M       1,000+  Free     0   
10839     4.5      114  Varies with device       1,000+  Free     0   
10840     4.5   398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [6]:
df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10840 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  int64  
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10841 non-null  object 
 9   Genres          10840 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10839 non-null  object 
dtypes: float64(1), int64(1), object(11)
memory usage: 1.1+ MB


In [8]:
df.describe()

Rating       Reviews
count  9367.000000  1.084100e+04
mean      4.191513  4.441119e+05
std       0.515735  2.927629e+06
min       1.000000  0.000000e+00
25%       4.000000  3.800000e+01
50%       4.300000  2.094000e+03
75%       4.500000  5.476800e+04
max       5.000000  7.815831e+07

## 3.11 `Observations:`

In [9]:
# showing all rows
pd.set_option('display.max_rows', None)

In [10]:
# showing unique values in each column with df.head()
df.columns.to_series().apply(lambda x: df[x].nunique())

App               9660
Category            33
Rating              39
Reviews           6001
Size               461
Installs            21
Type                 2
Price               92
Content Rating       6
Genres             119
Last Updated      1377
Current Ver       2831
Android Ver         33
dtype: int64

` Following are some anomalies in the data`
1. Rating has a maximum value of 19
2. Size has 'Varies with device',and  it should be numeric
3. Installs has '+' and ',', it should be numeric
4. Type has 'Free' and 'Paid'
5. Price has '$' and 'Everyone', it should be numeric
6. Content Rating has 'Adults only 18+' and 'Unrated'
7. Genres has ';'

# 4.0 `Exploring and Dealing with Columns:`

## 4.1 `App`:
### 4.1.1 `Exploring:`

In [11]:
df.App.isnull().sum()

0

In [12]:
df.App.nunique()

9660

In [13]:
# check for duplicates
df.duplicated().sum()

483

In [28]:
# Datatype
df.dtypes

App                object
Category           object
Rating            float64
Reviews             int64
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

### 4.1.2 `Observations:`

My observations about app column are:
 1. There are 483 duplicates
 2. There is no missing value
 2. There are 9660 unique values
 3. The data type is object


### 4.1.3 `Dealing With Duplicates::`

In [32]:
df[df.App.duplicated()]

App             Category  \
229                         Quick PDF Scanner + OCR FREE             BUSINESS   
236                                                  Box             BUSINESS   
239                                   Google My Business             BUSINESS   
256                                  ZOOM Cloud Meetings             BUSINESS   
261                            join.me - Simple Meetings             BUSINESS   
265                                                  Box             BUSINESS   
266                                             Zenefits             BUSINESS   
267                                           Google Ads             BUSINESS   
268                                   Google My Business             BUSINESS   
269                                                Slack             BUSINESS   
270                                   FreshBooks Classic             BUSINESS   
271                                        Insightly CRM             BUSINESS   
272          QuickBooks Accounting: Invoicing & Expenses             BUSINESS   
273                       HipChat - Chat Built for Teams             BUSINESS   
274                             Xero Accounting Software             BUSINESS   
275              MailChimp - Email, Marketing Automation             BUSINESS   
276                 Crew - Free Messaging and Scheduling             BUSINESS   
277                        Asana: organize team projects             BUSINESS   
278                                     Google Analytics             BUSINESS   
279                                      AdWords Express             BUSINESS   
280                          Accounting App - Zoho Books             BUSINESS   
281                       Invoice & Time Tracking - Zoho             BUSINESS   
282                            join.me - Simple Meetings             BUSINESS   
283    Invoice 2go — Professional Invoices and Estimates             BUSINESS   
284     SignEasy | Sign and Fill PDF and other Documents             BUSINESS   
285                         Quick PDF Scanner + OCR FREE             BUSINESS   
286                            Genius Scan - PDF Scanner             BUSINESS   
287                       Tiny Scanner - PDF Scanner App             BUSINESS   
288                         Fast Scanner : Free PDF Scan             BUSINESS   
289                     Mobile Doc Scanner (MDScan) Lite             BUSINESS   
290        TurboScan: scan documents and receipts in PDF             BUSINESS   
291                       Tiny Scanner Pro: PDF Doc Scan             BUSINESS   
292                        Docs To Go™ Free Office Suite             BUSINESS   
293               OfficeSuite : Free Office + PDF Editor             BUSINESS   
294                                                Slack             BUSINESS   
296          QuickBooks Accounting: Invoicing & Expenses             BUSINESS   
381                                   WhatsApp Messenger        COMMUNICATION   
382             Messenger – Text and Video Chat for Free        COMMUNICATION   
383                        imo free video calls and chat        COMMUNICATION   
385                                      Viber Messenger        COMMUNICATION   
386                                             Hangouts        COMMUNICATION   
390                                               WeChat        COMMUNICATION   
392                                             Telegram        COMMUNICATION   
393                                                  Who        COMMUNICATION   
394                                         Google Voice        COMMUNICATION   
396                                     Android Messages        COMMUNICATION   
408                   Firefox Focus: The privacy browser        COMMUNICATION   
410                                          Google Allo        COMMUNICATION   
411                         Google Chrome: Fast & Secure        COMMUNICATION   
412

## 4.2 `Category`:

## 4.3 `Rating`:

## 4.4 `Reviews`:

## 4.5 `Size`:

## 4.6 `Installs`:

## 4.7 `Type`:

## 4.8 `Price`:

## 4.9 `Content Rating`:

## 4.10 `Genres`:

## 4.11 `Last Updated`:

## 4.12 Current Version`:

## 4.13 `Andriod Version`: